core/state/access_witness.go

```diff
@@ -92,40 +92,40 @@ func (aw *AccessWitness) Copy() *AccessWitness {
 func (aw *AccessWitness) TouchFullAccount(addr []byte, isWrite bool) uint64 {
```

Comment:
I wonder if might be better to let the `AccessWitness` have access to `StateDB` to resolve if a write is or isn't filling.

As in, any time an `isWrite` (which is an existing parameter in stuff) has to charge gas, then ask `StateDB`: "Is this empty in the state?", if "Yes" charge, if "No" don't.

This avoids this new `isFill` parameter, avoids all callers having to send the right value, and maybe other risks?

Sounds like a natural design but maybe you've discarded that for reasons.
But look the further comments I did in below to see some risks I see with the current strategy.